File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
       int currentMin = prices[0];
       int maxProfit = 0;
       for(int i = 1; i < prices.length; i++) {
           if(prices[i] > currentMin) {
               maxProfit = Math.max(maxProfit,prices[i] - currentMin);
           } else {
               currentMin = prices[i];
           }
       }
        return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        min_start_cost = float("inf")
        max_profit = 0
        for price in prices:
            if min_start_cost < price:
                max_profit = max(max_profit,price - min_start_cost)
            min_start_cost = min(min_start_cost,price)
        
        return max_profit
        